src/sql-parser/src/parser.rs

```diff
@@ -3567,6 +3570,72 @@ impl<'a> Parser<'a> {
         ))
     }
 
+    fn parse_create_continual_task(&mut self) -> Result<Statement<Raw>, ParserError> {
+        // TODO(ct): If exists.
+        self.expect_keywords(&[CONTINUAL, TASK])?;
+
+        // TODO(ct): Multiple outputs.
+        let name = self.parse_item_name()?;
+        self.expect_token(&Token::LParen)?;
+        let columns = self.parse_comma_separated(|parser| {
+            Ok(CteMutRecColumnDef {
+                name: parser.parse_identifier()?,
+                data_type: parser.parse_data_type()?,
+            })
+        })?;
+        self.expect_token(&Token::RParen)?;
+        let in_cluster = self.parse_optional_in_cluster()?;
+
+        // TODO(ct): Multiple inputs.
+        self.expect_keywords(&[ON, INPUT])?;
+        let input_table = self.parse_raw_name()?;
+        // TODO(ct): Allow renaming the inserts/deletes so that we can use
+        // something as both an "input" and a "reference".
+
+        self.expect_keyword(AS)?;
+
+        let mut stmts = Vec::new();
+        let mut expecting_statement_delimiter = false;
+        self.expect_token(&Token::LParen)?;
+        // TODO(ct): Dedup this with parse_statements?
```

Comment:
+1 it seems like `parse_statements(...)` would be a drop in replacement? Was there anything you ran into initially where `parse_statements` didn't work?